_posts/2019-04-02-running-IMA-on-keylime.markdown

```diff
@@ -0,0 +1,441 @@
+---
+layout: blog
+title:  "Overview of Keylimes Runtime Integrity Management"
+author: "Luke Hinds"
+date:   2019-04-02
+---
+
+A feature of Keylime is runtime integrity monitoring, using the integrity management architecture (IMA).
+
+## Intro to IMA
+
+IMA was introduced to the Linux Kernel on the 20th May 2005 for the Kernel release 2.6.30 as part of the overall
+Linux Integrity Subsystem. The design was based upon the work of the open standard for trusted compute by the Trusted Computing Group (TCG).
+
+IMA (also known as the kernel integrity subsystem) provides a means of detecting whether files have been accidentally or maliciously altered or executed both remotely or locally.
+
+The following modules provide the integrity functions:
+
+* Collect – measure a file before it is accessed.
+* Store – add the measurement to a kernel resident list and, if a
+hardware Trusted Platform Module (TPM) is present, extend the IMA
+PCR
+* Attest – if present, use the TPM to sign the IMA PCR value, to
```

Comment:
Note that IMA itself is not doing the attesting, but writes the TPM PCR values necessary to be able to attest runtime state

Reply:
This is just a copy / paste from the IMA readme. Happy to change if you could give me that one sentence reworded.